### DISCUSSION TITLE
How to use ltree with JS SDK

### DISCUSSION THREAD

**Discussion by Shaun-Regenbaum** [supabase/supabase#18505] [Questions]
I have activated ltree extension to enable tree structures. I was wondering if there were any resources on best practices for using ltree with supabase's JS sdk? Specifically, some examples on how to traverse the tree structure?

**Comment by GaryAustin1** [comment]
https://github.com/PostgREST/postgrest/issues/2064
Talks about PostgREST supporting it and shows the URL format to make it work.  
https://postgrest.org/en/stable/references/api/tables_views.html?highlight=match#pattern-matching 
Is the query mentioned.

To do this from the JS sdk you would need to use the .filter() filter which lets you post PostgREST filters directly.
https://supabase.com/docs/reference/javascript/filter

Not tested.

**Comment by psychomelet** [comment]
`.filter("path", "match", "*.abc.*")` not working.

```json
{
  "error": {
    "code": "42883",
    "details": null,
    "hint": "No operator matches the given name and argument types. You might need to add explicit type casts.",
    "message": "operator does not exist: extensions.ltree ~ unknown"
  }
}
```

Tried others operators but none works.

Anyone get it work?

cc  @GaryAustin1

**Comment by gldkhoward** [comment]
Had this same problem, notes above on using filter worked. Was able to get child paths using this approach:

 
```
const descendantPattern = `${parent.path}.*`;

 const { data, error } = await this.supabase
      .from('locations')
      .select('id, path')
      .eq('is_active', true)
      .filter('path', 'match', descendantPattern); 
```

Docs from postgres: [https://www.postgresql.org/docs/current/ltree.html](https://www.postgresql.org/docs/current/ltree.html)
